src/components/schema/_schema_item.scss

```diff
@@ -0,0 +1,79 @@
+/*!
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+.ouiSchemaItem {
+  display: flex;
+  // sass-lint:disable-block no-misspelled-properties
+  gap: $ouiSizeS;
+  align-items: center;
+  padding: $ouiSizeS;
+  border-radius: $ouiBorderRadius;
+
+  &:hover {
+    @include ouiFocusBackground($ouiTextColor);
+
+    .ouiSchemaItem__actions .ouiButtonIcon {
+      opacity: 1;
+      visibility: visible;
+    }
+  }
+}
+
+.ouiSchemaItem__icon {
+  align-self: flex-start;
+  margin: $ouiSize / 2 0; // To align with buttonIcon
```

Comment:
this division will need to be surrounded by `calc()`, for sass upgrade

Reply:
This feels like a little bit of a hack? I think there are some existing approaches to align with buttons, although maybe I'm just thinking of https://oui.opensearch.org/1.0/#/navigation/button#empty-button

Reply:
@joshuarrrr I know why you feel that way, because i felt it too when i was implementing it. Open to suggestions here and probably can be improved in a followup, but the root of the problem here is that ButtonIcons have an margin that we need to match and unfortunately this is the best way i could find to do it in both the compressed and uncompressed mode.